tests/vertexonly/test_vertex_only_mesh_generation.py

```diff
@@ -0,0 +1,171 @@
+    creation coordinates `inputvertexcoords` behaves as expected.
+    `inputvertexcoords` should be the same for all MPI ranks to avoid
+    hanging.
+    """
+    gdim = m.geometric_dimension()
+    # Correct dims
+    assert vm.geometric_dimension() == gdim
+    assert vm.topological_dimension() == 0
+    # Can initialise
+    vm.init()
+    # Find in-bounds and non-halo-region input coordinates
+    in_bounds = []
+    _, owned, _ = m.cell_set.sizes
+    for i in range(len(inputvertexcoords)):
+        cell_num = m.locate_cell(inputvertexcoords[i])
+        if cell_num is not None and cell_num < owned:
+            in_bounds.append(i)
+    # Correct coordinates (though not guaranteed to be in same order)
+    np.allclose(np.sort(vm.coordinates.dat.data_ro), np.sort(inputvertexcoords[in_bounds]))
+    # Correct parent topology
+    assert vm._parent_mesh is m
+    assert vm.topology._parent_mesh is m.topology
+    # Correct generic cell properties
+    assert np.shape(vm.cell_closure) == (len(inputvertexcoords[in_bounds]), 1)
+    with pytest.raises(AttributeError):
+        vm._facets("")
+    with pytest.raises(AttributeError):
+        vm.cell_to_facets
+    assert vm.num_cells() == len(inputvertexcoords[in_bounds]) == vm.cell_set.size
+    assert vm.num_facets() == 0
+    assert vm.num_faces() == vm.num_entities(2) == 0
+    assert vm.num_edges() == vm.num_entities(1) == 0
+    assert vm.num_vertices() == vm.num_entities(0) == vm.num_cells()
+    # Correct parent cell numbers
+    stored_vertex_coords = np.copy(vm._swarm.getField("DMSwarmPIC_coor")).reshape((vm.num_cells(), gdim))
+    vm._swarm.restoreField("DMSwarmPIC_coor")
+    stored_parent_cell_nums = np.copy(vm._swarm.getField("parentcellnum"))
+    vm._swarm.restoreField("parentcellnum")
+    assert len(stored_vertex_coords) == len(stored_parent_cell_nums)
+    for i in range(len(stored_vertex_coords)):
+        assert m.locate_cell(stored_vertex_coords[i]) == stored_parent_cell_nums[i]
+
+
+def test_generate_cell_midpoints(parentmesh):
+    """
+    Generate cell midpoints for mesh parentmesh and check they lie in
+    the correct cells
+    """
+    inputcoords, inputcoordslocal = cell_midpoints(parentmesh)
+    vm = VertexOnlyMesh(parentmesh, inputcoords)
+    # Midpoints located in correct cells of parent mesh
+    V = VectorFunctionSpace(parentmesh, "DG", 0)
+    f = Function(V).interpolate(SpatialCoordinate(parentmesh))
+    # Check size of biggest len(vm.coordinates.dat.data_ro) so
+    # locate_cell can be called on every processor
+    max_len = MPI.COMM_WORLD.allreduce(len(vm.coordinates.dat.data_ro), op=MPI.MAX)
+    out_of_mesh_point = np.full((1, parentmesh.geometric_dimension()), np.inf)
+    for i in range(max_len):
+        if i < len(vm.coordinates.dat.data_ro):
+            cell_num = parentmesh.locate_cell(vm.coordinates.dat.data_ro[i])
+        else:
+            cell_num = parentmesh.locate_cell(out_of_mesh_point)  # should return None
```

Comment:
Can you open an issue tracking that `locate_cell` is collective even though it shouldn't be?

Reply:
See #1756